src/main/scala/lc/core/captchaProviders.scala

```diff
@@ -33,9 +37,26 @@ object CaptchaProviders {
     return providers(id)
   }
 
-  def getProvider(): ChallengeProvider = {
-    val keys = providers.keys
-    val providerIndex = keys.toVector(getNextRandomInt(keys.size))
-    providers(providerIndex)
+  private def filterProviderByParam(param: Parameters): List[(String, String)] = {
+    for {
+      JObject(child) <- config
+      JField("name", JString(name)) <- child
+      JField("supportedLevels", JArray(supportedLevels)) <- child
+      JField("supportedMedia", JArray(supportedMedia)) <- child
+      JField("supportedinputType", JArray(supportedinputType)) <- child
+      JField("config", config) <- child
```

Comment:
Parsing the JSON for every request would be very slow. You can parse it once and store all these fields somewhere:
`name`, `supportedLevels`, etc. Also, better to convert these to primitive types, so that you don't have to match them against JSON types like `JString(param.level)`